common/k8s.go

```diff
@@ -0,0 +1,62 @@
+package common
+
+import (
+	"flag"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// KubeConfig returns the kubeconfig path from either the KUBECONFIG
+// environment variable, the user's home directory, or the --kubeconfig
+// flag. Call this function **before** calling flag.Parse().
+func KubeConfig() *string {
+	var kubeconfig *string
+
+	if cluster := os.Getenv("CLUSTER"); cluster != "" {
+		return flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+
+	// Prefer the value in the KUBECONFIG env var.
+	// If the value is not set, then check for the HOME directory.
+	// If that is not set, then require the user to specify a path.
+	if kubeconfigEnv := os.Getenv("KUBECONFIG"); kubeconfigEnv != "" {
+		kubeconfig = flag.String("kubeconfig", kubeconfigEnv, "absolute path to the kubeconfig file")
+	} else if home := os.Getenv("HOME"); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		// If the home directory doesn't exist, then allow the user to specify a path.
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+
+	return kubeconfig
+}
+
+// RESTConfig returns the rest config needed by most k8s-related clients.
+// Call this **after** flag.Parse().
+func RESTConfig(kubeconfig string) (*rest.Config, error) {
```

Comment:
I would almost be tempted to remove the public `KubeConfig` function and have this function determine the path to the Kubernetes configuration file based on the value that's passed in and the environment variables (possibly by calling another private function).

Reply:
I'll have to completely change how the kubeconfig is set up. I'm not unwilling to do that, but I'd prefer that to be in another PR. As it is, this code needs to be called after  the flag.Parse() call and the other function has to be called before the flag.Parse() call.

Reply:
No worries. It was just a suggestion.